File: Data Structures & Algorithms/three-integer-sum/submission-0.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) 
    {
        HashSet<List<Integer>> ans = new HashSet<>();
        Arrays.sort(nums);
        for(int i = 0;i<nums.length;i++){
            int j = i+1;
            int k = nums.length-1;
            while(j<k){
                if(nums[j]+nums[k] == -nums[i]){
                    List<Integer> l = new ArrayList<>();
                    l.add(nums[i]);
                    l.add(nums[j]);
                    l.add(nums[k]);
                    ans.add(l);
                    j++;
                    k--;
                }
                else if(nums[j]+nums[k] < -nums[i]){
                    j++;
                }
                else{
                    k--;
                }
            }

        }

        return new ArrayList<>(ans);
    }
}

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.py
class Solution:
    def helper(self, nums, start, target, current_set, ans):
        if target == 0:
            ans.append(current_set[:])
            return
        if start == len(nums):
            return
        for i in range(start, len(nums)):
            if i > start and nums[i] == nums[i-1]:
                continue
            if nums[i] > target:
                break
            current_set.append(nums[i])
            self.helper(nums, i+1, target-nums[i], current_set, ans)
            current_set.pop()
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        candidates.sort()
        ans = []
        self.helper(candidates, 0, target, [], ans)
        return ans
        